tests/test_frameworks.py

```diff
@@ -0,0 +1,180 @@
+from contextlib import contextmanager
+from copy import deepcopy
+from itertools import chain
+import json
+
+from nose.tools import eq_
+import nose
+
+from js_helper import _do_real_test_raw as _js_test
+from validator import constants
+from validator.errorbundler import ErrorBundle
+from validator.testcases import content
+from validator.testcases.frameworks import fake_version
+
+
+KNOWN_SCRIPT = ("This file is a false script to facilitate testing of library "
+                "blacklisting.")
+
+
+def _do_test(xpi_package):
+    err = ErrorBundle()
+    content.test_packed_packages(err, xpi_package)
+
+    print err.print_summary(verbose=True)
+    return err
+
+
+class MockXPI(object):
+
+    def __init__(self, resources):
+        self.resources = resources
+
+    def info(self, name):
+        # Um...
+        return {"extension": name.split(".")[-1].lower()}
+
+    def read(self, name):
+        if isinstance(self.resources[name], bool):
+            return ""
+        if isinstance(self.resources[name], (dict, list, tuple)):
+            return json.dumps(self.resources[name])
+        return self.resources[name]
+
+    def __iter__(self):
+        for name in self.resources.keys():
+            yield name
+
+    def __contains__(self, name):
+        return name in self.resources
+
+
+@contextmanager
+def accepted_version(*path):
+    """
+    Temporarily marks the library, framework, or version at the given path
+    as accepted. Used mainly for tests on now-banned, outdated Jetpack
+    versions.
+    """
+    dict_ = constants.LIBRARY_METADATA
+    for elem in path:
+        dict_ = dict_[elem]
+
+    unset = object()
```

Comment:
Why use this? It doesn't seem idiomatic, and if it's a boolean, why not `dict_.get("banned", False)`?